core/src/main/scala/tech/beshu/ror/api/ConfigApi.scala

```diff
@@ -156,16 +151,56 @@ object ConfigApi {
       case object UpdateIndexConfig extends Type
     }
   }
-  
+
   sealed trait ConfigResponse
   object ConfigResponse {
-    def notAvailable: ConfigResponse = Failure("Service not available")
-    def internalError: ConfigResponse = Failure("Internal error")
+    sealed trait ForceReloadConfig extends ConfigResponse
+    object ForceReloadConfig {
+      final case class Success(message: String) extends ForceReloadConfig
+      final case class Failure(message: String) extends ForceReloadConfig
+    }
+
+    sealed trait ProvideIndexConfig extends ConfigResponse
+    object ProvideIndexConfig {
+      final case class Config(rawConfig: String) extends ProvideIndexConfig
+      final case class ConfigNotFound(message: String) extends ProvideIndexConfig
+      final case class Failure(message: String) extends ProvideIndexConfig
+    }
+
+    sealed trait ProvideFileConfig extends ConfigResponse
+    object ProvideFileConfig {
+      final case class Config(rawConfig: String) extends ProvideFileConfig
+      final case class Failure(message: String) extends ProvideFileConfig
+    }
+
+    sealed trait UpdateIndexConfig extends ConfigResponse
+    object UpdateIndexConfig {
+      final case class Success(message: String) extends UpdateIndexConfig
+      final case class Failure(message: String) extends UpdateIndexConfig
+    }
+
+    sealed trait Failure extends ConfigResponse
+    object Failure {
+      final case class BadRequest(message: String) extends Failure
+    }
+  }
+
+  implicit class StatusFromConfigResponse(val response: ConfigResponse) extends AnyVal {
```

Comment:
I left old status names. This could be improved in the future along with the swagger file.